app/javascript/components/widget-controller.jsx

```diff
@@ -65,19 +79,38 @@ export class WidgetController extends React.Component {
 
   moveDown = () => {
     this.setState(({ index }) => ({
-      index: (index + 1) % widgets.length,
+      index: (index + 1) % this.visibleWidgets.length,
     }));
   };
 
   moveUp = () => {
     this.setState(({ index }) => ({
-      index: mathMod(index - 1, widgets.length),
+      index: mathMod(index - 1, this.visibleWidgets.length),
     }));
   };
 
+  setVisibleWidgets = () => {
+    const date = new Date();
+    const currentHour = date.getHours();
+
+    return widgets.filter(widget => {
```

Comment:
@theroyalblah Since it's based on the _hour_, I think renaming `timeStart` and `timeEnd` to `hourStart` and `hourEnd` would make the code more self-documenting.

Also, since `0` is false-y, `if (widget.timeStart) ` will fail if the start time is midnight (`0`).

I think you should be able to rewrite this using a single line and enhance clarity (also, consider destructuring the `widget` props you use).

Reply:
Oh other pet peeve. Start/Stop or Begin/End. Never mixed.